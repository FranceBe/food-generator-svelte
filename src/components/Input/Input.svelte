<script xmlns="http://www.w3.org/1999/html">
    import H2 from 'components/H2/H2.svelte'

    export let label
    export let onChange
</script>
<style>
    label {
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;
        height: 70%;
        justify-content: center;
    }
    input {
        margin-left: 10px;
        height: 30px;
        width: 50%;
    }

    input:focus {
        outline: #EBA789 auto 5px;
    }

</style>

    <label>
        <H2 title={label}/>
        <input name="ingredients" on:change={onChange}/>
    </label>
