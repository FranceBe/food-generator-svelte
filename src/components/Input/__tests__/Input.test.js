import { render, fireEvent, cleanup } from '@testing-library/svelte'
import Input from 'components/Input/Input.svelte'

describe('Input', () => {
  afterEach(() => {
    cleanup()
  })
  it('should match snapshot', () => {
    const { container } = render(Input, {
      props: {
        label: 'My label',
        onChange: () => {
        },
      }
    })
    expect(container).toMatchSnapshot()
  })
  it('should render an input and a h2 tag', () => {
    const { container, getAllByText } = render(Input, {
      props: {
        label: 'My label',
        onChange: () => {
        },
      }
    })
    expect(container.getElementsByTagName('h2')).toHaveLength(1)
    expect(container.getElementsByTagName('input')).toHaveLength(1)
    expect(getAllByText('My label').length).toBe(1)
  })
  it('should call the props onChange when on:change is triggeres', async () => {
    const props = {
      onChange: jest.fn(),
      label: 'My label',
    }
    const { getByLabelText } = render(Input, {
      props,
    })
    const input = getByLabelText('My label')
    await fireEvent.change(input, { target: { value: 'tomatoe' } })

    expect(props.onChange).toHaveBeenCalledTimes(1)
  })
})
