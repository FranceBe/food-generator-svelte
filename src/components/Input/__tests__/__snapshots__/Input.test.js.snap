// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<body>
  <div>
    <label>
      <h2>
        My label
      </h2>
       
      <input
        name="ingredients"
      />
    </label>
  </div>
</body>
`;
